.github/workflows/check-for-new-contributor.yml

```diff
@@ -16,7 +16,12 @@ jobs:
   greet-first-time-pr:
     name: Greet First-Time PR Contributors
     runs-on: ubuntu-latest
-    if: github.event_name == 'pull_request_target' && github.event.action == 'opened' && github.event.pull_request.head.repo.fork
+    if: |
+      github.event_name == 'pull_request_target' && 
+      github.event.action == 'opened' && 
+      github.event.pull_request.head.repo.fork == true &&   # Ensure the PR is from a fork
+      !contains(secrets.CONTRIBUTOR_WHITELIST, github.actor)  # Check if the actor is not in the whitelist
```

Comment:
Define the whitelist as an environment variable that is defined in the workflow.
Use the git log to list all existing contributors since November 2024.
The definition in a secret cannot be maintained and there is no reason to hide the contributors.